### DISCUSSION TITLE
Highlight text via tags/badges/chips

### DISCUSSION THREAD

**Discussion by ciriousjoker** [outline/outline#3793] [Ideas]
I want to display tags like these inside tables:

![image](https://user-images.githubusercontent.com/16866223/179323464-bd621b76-c896-443f-b283-edf4f854aec0.png)

Basically, let me put arbitrary text into a small highlighted container. I'd like to pick from Material Design's 19 main (500) colors.

**Use case:**

We have a table where each row has a rating in respect to code quality. We have "good", "bad", "ok". Right now we upload and copy around small svg badges, but that's kind of annoying especially given the "download svg button"-overlay is almost as big as the 32x32 image.

Alternative solution:
- Upload custom "badges" in the settings and use them in several places via `/badge`
- Replacing an uploaded image should also replace all existing references to it.

**Comment by adrfantini** [comment]
This would be very useful, and especially useful if the chips act also as Google Docs dropdowns

**Reply by tommoor** [reply]
I'm not sure how you mean, could you explain or show screenshots of your expectations- thanks

**Reply by adrfantini** [reply]
There's nothing simpler! Go to Google Docs -> Insert -> Dropdown
You can then insert levels and get something like this:

<img width="343" height="206" alt="image" src="https://github.com/user-attachments/assets/09a6a2fb-c99c-4adf-98bf-f3f13277af34" />

This is very useful when creating templates for documents which include tags in the template. For example, you could use these to have a "Document state" dropdown tag, to quickly indicate below the title its state, or a "Company area", or anything that can be easily categorised and benefits from a quick glance.

**Reply by adrfantini** [reply]
Would this be something there might be some interest in @tommoor ?

**Reply by tommoor** [reply]
The on-premise version of Outline includes a similar "data attributes" feature that allows defining structured data for documents. The plan is to bring this to other versions of the product, I think it would overall end up as a better solution to the problem you're having here

**Reply by adrfantini** [reply]
Oh, interesting! I did not know that, is there anywhere where it is documented or where it can be tested?
